lib/archethic/mining/pending_transaction_validation.ex

```diff
@@ -671,8 +671,71 @@ defmodule Archethic.Mining.PendingTransactionValidation do
     end
   end
 
+  @code_max_size Application.compile_env!(:archethic, :transaction_data_code_max_size)
+  defp do_accept_transaction(%Transaction{type: :contract, data: %TransactionData{code: code}}, _) do
+    cond do
+      byte_size(code) == 0 ->
+        {:error, "invalid contract type transaction -  code is empty "}
+
+      byte_size(code) >= @code_max_size ->
+        {:error, "invalid contract type transaction , code exceed max size "}
+
+      true ->
```

Comment:
can we check that it's parse-able? `Archethic.Contracts.Interpreter.parse(code)`

Reply:
we are already doing that under validate_contract , line 256.